crates/goose/src/session/session_manager.rs

```diff
@@ -895,6 +903,41 @@ impl SessionStorage {
             total_tokens: row.1.unwrap_or(0),
         })
     }
+
+    async fn export_session(&self, id: &str) -> Result<String> {
+        let session = self.get_session(id, true).await?;
+        serde_json::to_string_pretty(&session).map_err(Into::into)
+    }
+
+    async fn import_session(&self, json: &str) -> Result<Session> {
+        let import: Session = serde_json::from_str(json)?;
+
+        let session = self
+            .create_session(import.working_dir.clone(), import.description.clone())
+            .await?;
+
+        self.apply_update(
+            SessionUpdateBuilder::new(session.id.clone())
+                .extension_data(import.extension_data)
+                .total_tokens(import.total_tokens)
+                .input_tokens(import.input_tokens)
+                .output_tokens(import.output_tokens)
+                .accumulated_total_tokens(import.accumulated_total_tokens)
+                .accumulated_input_tokens(import.accumulated_input_tokens)
+                .accumulated_output_tokens(import.accumulated_output_tokens)
+                .schedule_id(import.schedule_id)
+                .recipe(import.recipe)
+                .user_recipe_values(import.user_recipe_values),
```

Comment:
consider constructing the struct itself without the builder so any added new fields are caught at compile time

Reply:
I had that at first, but it becomes awkward since we still want to call create_session at this point so then we need to update only the other fields?